src/sqlfluff/cli/commands.py

```diff
@@ -365,6 +367,26 @@ def lint(
         click.echo(json.dumps(result.as_records()))
     elif format == "yaml":
         click.echo(yaml.dump(result.as_records()))
+    elif format == "github-annotation":
+        github_result = []
+        for record in result.as_records():
+            filepath = record["filepath"]
+            for violation in record["violations"]:
+                # NOTE: The output format is designed for this GitHub action:
+                # https://github.com/yuzutech/annotations-action
+                # It is similar, but not identical, to the native GitHub format:
+                # https://docs.github.com/en/rest/reference/checks#annotations-items
+                github_result.append(
+                    {
+                        "file": filepath,
+                        "line": violation["line_no"],
+                        "start_column": violation["line_pos"],
+                        "title": "SQLFluff",
+                        "message": f"{violation['code']}: {violation['description']}",
+                        "annotation_level": "warning",
```

Comment:
would it make sense to make `annotation_level` configurable? in our case we'd want this to be annotated as a failure that blocks a merge vs just a warning

Reply:
Sure! Are you suggesting a command-line parameter that is passed through to every message (i.e. identical value), or something more complicated?

I see there's an existing issue (#201) for "something more complicated".

I think something like the "identical value" approach would be reasonable for this PR. It would be cool to eventually address #201, but I think that's probably out of scope for this PR.

What do you think, @emily-hawkins?

Reply:
i think a command line param would be fine! right now we're treating all rules the same anyways

Reply:
Sounds great! I'll try and add an `--annotation-level` option tomorrow, with choices:
* notice
* warning
* failure

(List of choices from the GitHub docs.)